src/testing/native-debugging.md

```diff
@@ -0,0 +1,458 @@
+---
+title: Use a native language debugger
+short-title: debuggers
+description: How to connect a native debugger to your running Flutter app.
+---
+
+<?code-excerpt path-base="testing/native_debugging"?>
+
+{{site.alert.note}}
+  This guide presumes you understand general debugging,
+  have installed Flutter and git, and have familiarity
+  with the Dart language as well as one of the following
+  languages: Java, Kotlin, Swift, or ObjectiveC.
+{{site.alert.end}}
+
+If you write Flutter apps only with Dart code,
+you can debug your code using your IDE's debugger.
+The Flutter team recommends VS Code.
+
+If you write a platform-specific plugin or
+using platform-specific libraries, you can debug
+that portion of your code with a native debugger.
+
+- To debug iOS or macOS code written in Swift or ObjectiveC,
+  you can use Xcode.
+- To debug Android code written in Java or Kotlin, you can use Gradle.
+- To debug Windows code written in C++, you can use Visual Studio.
+
+This guide shows you how you can connect _two_
+debuggers to your Dart app, one for Dart, and one for the native code.
+
+## Debug Dart code
+
+This guide describes how to use VS Code to debug your Flutter app.
+You can also use your preferred IDE with the
+Flutter and Dart plugins installed and configured.
+
+## Debug Dart code using VS Code
```

Comment:
This guide on debugging Dart code is excellent - should it be moved to [general debugging page](https://tony-flutter-site.web.app/testing/debugging)? It is a huge improvement over the current Flutter debugger docs, and moving it there would make it easier for folks to discover this content. Also, it feels out of place here as this page documents native debuggers.

Reply:
Yeah, I thought the same thing, but was willing to push this so it's somewhere and make a separate PR to move it.